plugins/filter/filters.py

```diff
@@ -32,46 +33,67 @@ def filters(self):
             'resolve_principal': self.resolve_principal
         }
 
-    def normalize_sasl_protocol(self, protocol):
-        # Returns standardized value for sasl mechanism string
-        normalized = 'GSSAPI' if protocol.lower() == 'kerberos' \
-            else 'SCRAM-SHA-512' if protocol.upper() == 'SCRAM' \
-            else 'SCRAM-SHA-256' if protocol.lower() == 'scram256' \
-            else 'PLAIN' if protocol.upper() == 'PLAIN' \
-            else 'OAUTHBEARER' if protocol.upper() == 'OAUTH' \
-            else 'none'
-        return normalized
-
-    def kafka_protocol_normalized(self, sasl_protocol_normalized, ssl_enabled):
+    def normalize_sasl_protocol(self, protocols):
+        # Returns a list of standardized values for sasl mechanism strings
+        protocol_list = protocols.replace(" ", "").split(',')
+
+        normalized_protocols = []
+        for protocol in protocol_list:
+            if protocol.lower() == 'kerberos':
+                normalized = 'GSSAPI'
+            elif protocol.upper() == 'SCRAM':
+                normalized = 'SCRAM-SHA-512'
+            elif protocol.lower() == 'scram256':
+                normalized = 'SCRAM-SHA-256'
+            elif protocol.upper() == 'PLAIN':
+                normalized = 'PLAIN'
+            elif protocol.upper() == 'OAUTH':
+                normalized = 'OAUTHBEARER'
+            else:
+                normalized = 'none'
+            normalized_protocols.append(normalized)
```

Comment:
Here if someone makes a mistake in their inventory file then that would cause this function to return a list something like ['none', 'OAUTHBEARER']. So that might later cause confusion wherever we are checking 'none' in normalize_sasl_protocols. 
Also another side point are there any pros/cons of returning 'none' vs an empty list ?

Reply:
We have added a validation to allow maximum 2 values and [none,oauth] are not allowed when providing multiple values.